README.md

```diff
@@ -4,27 +4,720 @@
+
+Your application may or may not need this depending on how you have 
+things set up (the fact that `subscribe!` returns `nil` if unauthorized
+may be enough for you). It is just provided as an extra convenience.
+
+## Namespaces
+
+As has been mentioned already, namespaces can be used to isolate 
+subscriptions to views and view refreshes. Typical use of namespaces 
+within a views system would be to set them to something that specifies 
+which database to retrieve view data from when you have multiple 
+databases all with an identical structure.
+
+Namespace information is not included in the actual view refresh data 
+that gets sent to subscribers. It is just considered to be a 
+server-side concern.
+
+Depending on your application, you may be perfectly ok with just 
+passing in the specific namespace needed when creating view 
+subscriptions. However, you can also specify a `:namespace-fn` option 
+in your call to `views.core/init!` and provide a function that will 
+return the namespace to use for all calls to `subscribe!` and 
+`unsubscribe!` that get passed a view sig which **does not** include a 
+namespace in it.
+
+The `:namespace-fn` function should be of the form:
+
+```clj
+(fn [view-sig subscriber-key context]
+  ; ...
+  )
+```
+
+`context` will be whatever was passed in as the context argument to 
+`subscribe!`/`unsubscribe!`.
+
+It bears repeating that `:namespace-fn` will **not** be called even if
+it was set if you use a view sig that includes a `:namespace` key.
+
+For this reason the helper function `->view-sig` includes an extra 
+overload that does not set a namespace.
+
+```clj
+; a view sig that will result in namespace-fn being called (if one is set)
+(views/->view-sig :foo [])
+=> {:view-id :foo, :parameters []}
+
+; a view sig that will always use :a as the namespace, even if a namespace-fn is set
+(views/->view-sig :a :foo [])
+=> {:namespace :a, :view-id :foo, :parameters []}
+```
+
+## View System Initialization Options
+
+There are a number of options that can be provided to 
+`views.core/init!`. The only one that absolutely must be provided for a
+working system is `:send-fn` while all the other default options will
+generally suffice for a non-distributed relatively low-load
+application.
+
+The default options are defined in `views.core/default-options`.
+
+#### `:send-fn`
+
+A function that is used to send view refresh data to subscribers.
+
+```clj
+(fn [subscriber-key [view-sig view-data]]
+  ; ...
+  )
+```
+
+#### `:views`
+
+A list of `IView` instances. These are the views that can be 
+subscribed to. Views can also be added/replaced in the system after 
+initialization by calling `views.core/add-views!`.
+
+#### `:put-hints-fn`
+
+A function that typically will be used by the different views plugin 
+libraries providing view implementations (such as 
+[views.sql](https://github.com/gered/views.sql) or
+[views.honeysql](https://github.com/gered/views.honeysql)) to add 
```

Comment:
Probably not until we update our honeysql repo. We don't actually use our repo in our system (we use a tweaked version that handles the firm database).